src/api/app/components/notification_component.rb

```diff
@@ -29,4 +29,76 @@ def notification_icon
       tag.i(class: ['fas', NOTIFICATION_ICON[@notification.notifiable_type]], title: NOTIFICATION_TITLE[@notification.notifiable_type])
     end
   end
+
+  def description
+    case @notification.event_type
+    when 'Event::ReportForUser'
+      description_for_user_report
+    when 'Event::ReportForComment'
+      description_for_comment_report
+    else
+      @notification.description
+    end
+  end
+
+  private
+
+  def description_for_user_report
+    reporter = @notification.notifiable.user
+    accused = @notification.notifiable.reportable
+    reports_on_comments = count_reports_on_comments(accused)
+    reports_on_user = count_reports_on_user(accused)
+
+    generate_report_description(reporter, accused, reports_on_comments, reports_on_user)
+  end
+
+  def description_for_comment_report
+    reporter = @notification.notifiable.user
+    accused = @notification.notifiable.reportable.user
+    reports_on_user = count_reports_on_user(accused)
+    reports_on_comments = count_reports_on_comments(accused)
+
+    generate_report_description(reporter, accused, reports_on_comments, reports_on_user, comment_link: @notification.link_path)
+  end
+
+  def count_reports_on_comments(accused)
+    Report.where(reportable: accused.comments, decision: nil).count
+  end
+
+  def count_reports_on_user(accused)
+    Report.where(reportable: accused, decision: nil).count
+  end
+
+  def generate_report_description(reporter, accused, reports_on_comments, reports_on_user, comment_link: nil)
+    text = link_to(reporter, user_path(reporter, notification_id: @notification.id))
+    text += ' created a report for '
+
+    if comment_link
```

Comment:
@rubhanazeem not sure if the an additional link to the comment is required. The the notification already links to the corresponding comment. It would reduce the amount of code. To me having the link twice feels redundant.

Reply:
Thanks for the suggestion. I've removed the additional link to comment